Document bumping version (#16195)
conan-io/conan-center-index: docs/bump_version.md

> # How to bump a package version for an existing recipe in ConanCenterIndex?

Reviewer: I don't think this is what the complaints are about...

I think people assume that updating a package (which is naively new versions) is different from adding a package.

I absolutely agree there's a problem but this seem more narrow than what is needed 🤔 

I would suggest a new doc "update existing recipes" and i would image a "adding new versions"  along with links to the other more details docs tie them together 

Just me thinking out loud

Author: > I would suggest a new doc "update existing recipes" and i would image a "adding new versions" along with links to the other more details docs tie them together

This would fit yes! But I would prefer step-by-step. Documentation is always massive to review, so small PRs are easier to digest.

Once a recipe is approved and merged to ConanCenterIndex, it may need to be updated with new versions released by the upstream.

> When someone needs a new package version that is not available in ConanCenterIndex, then open a new pull request adding. This is

Author: Fixed, thanks for reviewing.

> When someone needs a new package version that is not available in ConanCenterIndex, then open a new pull request adding, is what is

Reviewer: ```suggestion
When someone needs a new package version that is not available in ConanCenterIndex, then open a new pull request adding. This is
```

> called `bump version`.

Reviewer: ```suggestion
called a `bump version`.
```

> A bumping version process is when a PR only adds a new package version and nothing more. Removing older versions, or updating

Reviewer: ```suggestion
The [build service](#) bumping process is limited to pull requests which **only** adds a new package version and nothing more. Removing older versions, or updating
```

> the recipe is not classified as a bump version action.

Reviewer: ```suggestion
the recipe will disqualify a pull request from the `Bump version` [review process](#)
```

## Choosing which version should be updated

The first step is checking the version which should be added from the upstream. Please, avoid adding multiple versions which you do not
need. Adding a new version will increase the building time and storage for each new package configuration.

> Once you detect which version should be updated, please, first check if the project `license` keeps the same.

Reviewer 2: Newer versions might have more knots that Conan might need to handle (more dependencies, configuration options, etc) which might require modifying the recipe.
I think this situation should also be clarified and that due to this (as stated before) the PR won't be labeled as `bump version`.

Author: It's documented on https://github.com/conan-io/conan-center-index/blob/master/docs/labels.md#bump-version, but clarifying here is valid too.

Author: Done.

# How to bump a package version for an existing recipe in ConanCenterIndex?

Once a recipe is approved and merged to ConanCenterIndex, it may need to be updated with new versions released by the upstream.
When someone needs a new package version that is not available in ConanCenterIndex, that person can open an issue [resquesting a new version](https://github.com/conan-io/conan-center-index/issues/new?assignees=&labels=upstream+update&template=package_upstream_update.yml&title=%5Brequest%5D+%3CLIBRARY-NAME%3E%2F%3CLIBRARY-VERSION%3E).

> Or, by opening a pull request changing and adding that needed version, this is called a `bump version`.

Reviewer 3: ```suggestion
Or, by opening a pull request changing and adding that version needed, this is called a `bump version`.
```

> The [build service](adding_packages/README.md#the-build-service) bumping process is limited to pull requests which **only** adds a new package version and nothing more. Removing older versions, or updating

Reviewer 3: ```suggestion
The [build service](adding_packages/README.md#the-build-service) bumping process is limited to pull requests which **only** add a new package version and nothing more. Removing older versions, or updating
```

the recipe will disqualify a pull request from the `Bump version` [review process](review_process.md)

## Choosing which version should be updated

The first step is checking the version which should be added from the upstream. Please, avoid adding multiple versions which you do not
need. Adding a new version will increase the building time and storage for each new package configuration.

Once you detect which version should be updated, please, first check if the project `license` keeps the same.

> > In case needing to update attributes, dependencies versions, patches, or anything besides the package version in `conandata.yml` and `config.yml`,

Reviewer 3: ```suggestion
> In case you need to update attributes, dependencies versions, patches, or anything besides the package version in `conandata.yml` and `config.yml`,
```
